### DISCUSSION TITLE
check_run / check_suite events always run workflow against latest HEAD instead of checked commit

### DISCUSSION THREAD

**Discussion by chokoswitch** [community/community#25510] [Actions]
<p>Hello,</p>
<p>I am trying to setup an action that simply comments on a PR issue or commit once the check completes. However, I’ve found that workflows started from check_run and suite events do not run in the context of the commit that started the check. While it’s probably still possible to add a comment by hitting the GitHub API manually, it should be as easy as using </p>
<pre><code>actions/github comment Success!
</code></pre>
<p>And even if hitting API manually, the workflow result will be located in the wrong place and can cause confusion.</p>
<p><a href="https://github.com/curioswitch/curiostack/runs/354137253" rel="nofollow noopener">https://github.com/curioswitch/curiostack/runs/354137253</a> shows the behavior. You can see many Cloud Build Notifier workflow runs here, but only the top one is correct.</p>
<p>In <a href="https://github.com/curioswitch/curiostack/runs/354137253" rel="nofollow noopener">https://github.com/curioswitch/curiostack/runs/354137253</a></p>
<pre><code>github.ref == github.event.check_run.check_suite.head_sha == ae3edd085ec84b4421c38506a01b4b976686fbfb
</code></pre>
<p>But just below if you click on <a href="https://github.com/curioswitch/curiostack/runs/354133605" rel="nofollow noopener">https://github.com/curioswitch/curiostack/runs/354133605</a></p>
<pre><code>github.ref == ae3edd085ec84b4421c38506a01b4b976686fbfb
github.event.check_run.check_suite.head_sha ==d8f3f602db73eb534e2c229937c5808f39aab009
</code></pre>
<p>The check is a completely different commit but the workflow is run against master only.</p>
<p>Is this intended behavior? I would expect that github.ref to always be the same as check_suite.head_sha, meaning the workflow is always executed in the context of the checked commit.</p>
<p>Thanks.</p>

**Comment by chrispat** [comment, marked as answer]
<p>This is by design.  In the case of check_suite and check_run these are really API oriented events and not code oriented events and GitHub Actions always runs API events from the default branch and provides you all of the payload info and token so you can manipulate the API.  </p>

**Comment by chokoswitch** [comment]
<p>Thanks - yeah I thought it might be. I’m curious what the distinction between code oriented and API oriented is, since I thought checks are usually run against code, hence having a head_sha field in the payload. If it is what it is, I’ll go ahead and use the API directly, but I would consider a feature request to tie checks to their head_sha commit since it seems more natural to me.</p>
